### PR TITLE
Integration/unit tests for Stash/Bitbucket collectors/clients

### PR DESCRIPTION
**Pull request by srid99**
Added test cases for Stash and Bitbucket clients.

Do like to mention I found few things while writing the test case that there are few things should be fixed/improved. I added TODO's in the places where I found things has to be improved. I didn't fix them since I wanted to keep the implementation as it is and only getting the unit tests first so that we can refactor later (happily). Please go through the TODO's and let me know your opinion. Again I am fine with fixing them. Feel free to ping me in gitter if something is not clear or if I did something stupid (possibly)...

### CODE REVIEW

File: stash-scm-collector/pom.xml [hunk 1]
```diff
@@ -1,106 +1,107 @@
...
-	</properties>
+        <testResources>
+            <testResource>
+                <directory>src/test/resources</directory>
+            </testResource>
+        </testResources>
+    </build>
 
+    <properties>
+        <java.version>1.7</java.version>
```
Comment (jdamick) [inline stash-scm-collector/pom.xml]: we should be up on java8

File: stash-scm-collector/pom.xml [hunk 2]
```diff
@@ -1,106 +1,107 @@
...
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-mongodb</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-web</artifactId>
+            <version>4.1.0.RELEASE</version>
```
Comment (jdamick) [inline stash-scm-collector/pom.xml]: dont lock to a specific ver, i think the base as this as property

File: stash-scm-collector/src/test/java/com/capitalone/dashboard/collector/DefaultStashClientTest.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    // TODO Throwing NPE doesn't seem right. Maybe this scenario (implementation) has to be handled properly.
+    @Test(expected = NullPointerException.class)
+    public void unexpectedResponse() {
+        server.response("{}");
+
+        stashClient.getCommits(gitRepo(), true);
+    }
+
+    // TODO Throwing NPE doesn't seem right. Maybe this scenario (implementation) has to be handled properly.
+    @Test(expected = NullPointerException.class)
```
Comment (jdamick) [inline stash-scm-collector/src/test/java/com/capitalone/dashboard/collector/DefaultStashClientTest.java]: agreed, shouldnt be NPE

**Comment by MarkRx** [comment]
Does this add the ability to choose which implementation is used?

**Pull request closed without merging**